### PR TITLE
Add a glossary entry for the term 'coarse mesh'.

### PR DESCRIPTION
**Pull request by bangerth**
I'll reference this entry from various functions in namespace GridGenerator
in a follow-up.

### CODE REVIEW

File: doc/doxygen/headers/glossary.h [hunk 1]
```diff
@@ -377,7 +377,32 @@
...
+ *   of cells that are not refined, i.e., a mesh in which no cell is a child
+ *   of another cell. This is generally how triangulations are first
+ *   constructed in deal.II, for example using (most of) the functions in
+ *   namespace GridGenerator, the functions in class GridIn, or directly
+ *   using the function Triangulation::create_triangulation(). One can of
+ *   course do computations on such meshes, but most of the time (see, for
+ *   example, almost any of the tutorial programs) one first refines the
+ *   coarse mesh either globally (using Triangulation::refine_global()),
+ *   or adaptively using a refinement criterion, the functions in
+ *   namespace GridRefinement, and then calling
```
Comment (masterleinad) [inline doc/doxygen/headers/glossary.h]: I don't understand the part `[...] using a refinement criterion, the functions in namespace GridRefinement,[...]` syntactically.
Do you want to say something like `[...] using a refinement criterion (e.g, one of the ones implemented in namespace GridRefinement) [...]`?
Also, adaptive mesh refinement and global refinement are not mutually exclusive, so I would just skip "either".

**Comment by bangerth** [comment]
@masterleinad : I've rephrased the sentence in question. Do you like this better now?

**Comment by drwells** [comment]
I like this and I think its ready to merge aside from what @masterleinad mentioned.

I think this is the right place to mention that a coarse mesh only has cells on level 0. Would you mind adding, either here or in a follow-up, a sentence that says, roughly, "Triangulation objects store cells in different *levels*: in particular, all cells of a coarse Triangulation are on level zero. If we executed `Triangulation::refine_global(1)` on a coarse Triangulation then all cells would be at level one", My wording could be improved, but I think we should mention this since its something assertions check (I think) so its occasionally printed to the screen when I make mistakes.

**Comment by bangerth** [comment]
@drwells : I've added a paragraph like the one you suggested.

**Comment by bangerth** [comment]
Ping?

File: doc/doxygen/headers/glossary.h [hunk 2]
```diff
@@ -377,7 +377,42 @@
...
+ *   started out with, i.e., from which all the currently
+ *   @ref GlossActive "active cells" of the triangulation have been obtained
+ *   by mesh refinement. (Some of the coarse mesh cells may of course also
+ *   be active if they have never been refined.)
+ *
+ *   Triangulation objects store cells in <i>levels</i>: in
+ *   particular, all cells of a coarse mesh are on level zero. Their
+ *   children (if we executed Triangulation::refine_global(1) on a
+ *   coarse mesh) would then be at level one, etc. Consequently, a
+ *   coarse mesh only has cells on level zero. The coarse mesh of a
```
Comment (drwells) [inline doc/doxygen/headers/glossary.h]: This sentence (`Consequently, a coarse mesh only has cells on level zero`) is redundant: the first sentence already says `all cells of a coarse mesh are on level zero`,

**Comment by bangerth** [comment]
How about now?

**Comment by drwells** [comment]
:+1:

**Pull request merged**